app/Controllers/Http/AuthController.ts

```diff
@@ -0,0 +1,87 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import User from 'App/Models/User'
+import Hash from '@ioc:Adonis/Core/Hash'
+import Database from '@ioc:Adonis/Lucid/Database'
+import Joi from 'joi'
+export default class AuthController {
+
+    // public async login({ request, response ,auth}: HttpContextContract) {
+    //     const { email, password } = request.only(['email', 'password'])
+    //     let user = await User.findBy('email', email)
+    //     if (!user) {
+    //         response.abort('Invalid credentials')
+    //     }
+    //     await Hash.verify(user.password, password)
+
+    // }
+
+    public async login({ request, response, auth }: HttpContextContract) {
+        const { email, password } = request.only(['email', 'password'])
+        // Find the user by email
+
+        const user = await User.findBy('email', email)
+
+        // If user does not exist, return error
+        if (!user) {
+            return response.status(401).json({ error: 'Invalid credentials' })
+        }
+
+        // Verify password
+        const isPasswordValid = await Hash.verify(user.password, password)
+
+        // If password is not valid, return error
+        if (!isPasswordValid) {
+            return response.status(401).json({ error: 'Invalid credentials' })
+        }
+
+        // Generate token for the user
+        const token = await auth.use('api').generate(user, {
+            expiresIn: '7 days' // Set expiry for token
+        })
+        // return await auth.attempt(email, password)
+
+        // Return token
+        return response.status(200).json({ token })
+    }
+
+
+    public async register({ request, response }: HttpContextContract) {
+        await this.validation(request.body())
+
+        let isuserexist = await User.findBy('email', request.body().email)
+        if (isuserexist) {
+            return response.json({ message: "email already exist" })
+        }
+
+        const newUser = request.all() as Partial<User>
+        const hashnewpwd = await Hash.make(newUser.password)
+        let userobj = {
+            email: newUser.email,
+            password: hashnewpwd
+        }
```

Comment:
You can add hashing logic in model hook for password hashing

Ref: https://v5-docs.adonisjs.com/guides/security/hashing

```ts
@beforeSave()
  public static async hashPassword (user: User) {
    if (user.$dirty.password) {
      user.password = await Hash.make(user.password)
    }
  }
```